### PR TITLE
[#12048] Migrate GetFeedbackQuestionAction

### PR DESCRIPTION
**Pull request by cedricongjh**
Part of #12048 

**Outline of Solution**

As per title.

Added the following logic methods:
- `getFeedbackQuestionsForStudents`
- `getFeedbackQuestionsForInstructors`
- `populateFieldsToGenerateInQuestion`
- `getTeamsForSection`
- `getTeamsForCourse`
- `getStudentsForSection`
- `getStudentsForTeam`

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/GetFeedbackQuestionsAction.java [hunk 1]
```diff
@@ -102,4 +150,12 @@ public JsonResult execute() {
         return new JsonResult(response);
     }
 
+    private List<FeedbackQuestionAttributes> convertToFeedbackQuestionAttributes(List<FeedbackQuestion> questions) {
```
Comment (samuelfangjw) [inline src/main/java/teammates/ui/webapi/GetFeedbackQuestionsAction.java:153]: should not be using attributes class anymore.

Comment (cedricongjh) [reply src/main/java/teammates/ui/webapi/GetFeedbackQuestionsAction.java]: Alright, got it, I see that @hhdqirui's PR has done that, will pull from his branch once it's merged

**Review by hhdqirui — COMMENTED** [review]
LGTM! Just some nits

File: src/main/java/teammates/storage/sqlapi/CoursesDb.java [hunk 2]
```diff
@@ -145,6 +146,41 @@ public Section getSectionByCourseIdAndTeam(String courseId, String teamName) {
...
+        Join<Team, Section> teamJoin = teamRoot.join("section");
+
+        cr.select(teamRoot).where(
+                cb.equal(teamJoin.get("id"), section.getId()));
+
+        return HibernateUtil.createQuery(cr).getResultList();
+    }
+
+    /**
+     * Get team by {@code course}.
```
[the remark this answers is not in the document]
Comment (cedricongjh) [reply src/main/java/teammates/storage/sqlapi/CoursesDb.java]: changed!

File: src/main/java/teammates/storage/sqlapi/CoursesDb.java [hunk 3]
```diff
@@ -145,6 +146,41 @@ public Section getSectionByCourseIdAndTeam(String courseId, String teamName) {
         return HibernateUtil.createQuery(cr).getResultStream().findFirst().orElse(null);
     }
 
+    /**
+     * Get team by {@code section}.
```
[the remark this answers is not in the document]
Comment (cedricongjh) [reply src/main/java/teammates/storage/sqlapi/CoursesDb.java]: changed!

**Review by hhdqirui — APPROVED** [review]

**Comment by hhdqirui** [comment]
@samuelfangjw Do you have any other comments?

**Pull request merged**